atc/db/team_test.go

```diff
@@ -916,6 +916,128 @@ var _ = Describe("Team", func() {
 				Expect(len(workers)).To(Equal(0))
 			})
 		})
+
+		Context("when there is a second worker", func() {
+			var atcWorker2 = atc.Worker{
+				GardenAddr:       "some-garden-addr-2",
+				BaggageclaimURL:  "some-bc-url-2",
+				HTTPProxyURL:     "some-http-proxy-url-2",
+				HTTPSProxyURL:    "some-https-proxy-url-2",
+				NoProxy:          "some-no-proxy-2",
+				ActiveContainers: 140,
+				ResourceTypes: []atc.WorkerResourceType{
+					{
+						Type:    "some-base-resource-type",
+						Image:   "some-image",
+						Version: "some-version",
+					},
+				},
+				Platform:  "some-platform",
+				Name:      "some-worker-name-2",
+				StartTime: 55,
+			}
+
+			var volumeOnWorker2 db.CreatedVolume
+
+			BeforeEach(func() {
+				_, err := workerFactory.SaveWorker(atcWorker2, 0)
+				Expect(err).ToNot(HaveOccurred())
+
+				creatingVolume, err := volumeRepository.CreateVolume(defaultTeam.ID(), atcWorker2.Name, db.VolumeTypeResource)
+				Expect(err).ToNot(HaveOccurred())
+
+				volumeOnWorker2, err = creatingVolume.Created()
+				Expect(err).ToNot(HaveOccurred())
+
+				err = volumeOnWorker2.InitializeStreamedResourceCache(urc, atcWorker.Name)
+				Expect(err).ToNot(HaveOccurred())
+			})
+
+			Context("when the workers are running", func() {
+				BeforeEach(func() {
+					atcWorker.State = string(db.WorkerStateRunning)
+					_, err := workerFactory.SaveWorker(atcWorker, 0)
+					Expect(err).ToNot(HaveOccurred())
+
+					atcWorker2.State = string(db.WorkerStateRunning)
+					_, err = workerFactory.SaveWorker(atcWorker2, 0)
+					Expect(err).ToNot(HaveOccurred())
+				})
+
+				It("should find both workers", func() {
+					workers, err := defaultTeam.FindWorkersForResourceCache(urc.ID())
+					Expect(err).ToNot(HaveOccurred())
+					Expect(len(workers)).To(Equal(2))
+					Expect(workers[0].Name()).To(Equal("some-worker-name"))
+					Expect(workers[1].Name()).To(Equal("some-worker-name-2"))
+				})
+
+				Context("when worker1 is pruned", func() {
+					BeforeEach(func() {
+						_, err := psql.Delete("workers").Where(sq.Eq{"name": "some-worker-name"}).RunWith(dbConn).Exec()
```

Comment:
i think you could do workerFactory.GetWorker() and then worker.Prune(), refer to https://github.com/concourse/concourse/blob/185459d03b0b1b173c3fac81ea54539d9e680436/atc/api/workerserver/prune.go#L15